tests/rptest/tests/write_caching_fi_test.py

```diff
@@ -0,0 +1,380 @@
+        requests are acknowledged once they have been replicated to majority of
+        replicas which will form a new quorum after leader failure.
+        """
+
+        num_restart_rounds = 3
+        num_msg_per_round = 5000
+
+        total_produced = 0
+
+        for round_ix in range(0, num_restart_rounds + 1):
+            self.logger.info(f"Round {round_ix}/{num_restart_rounds}")
+
+            if round_ix > 0:
+                self._crash_and_restart_leader(wait_for_new_leader=True)
+
+                hwm = next(self.rpk.describe_topic(self.topic)).high_watermark
+                if total_produced != hwm:
+                    raise Exception(
+                        f"Lost messages: {total_produced - hwm} in round {round_ix}"
+                    )
+
+            KgoVerifierProducer.oneshot(self.test_context, self.redpanda,
+                                        self.topic_name, self.MSG_SIZE,
+                                        num_msg_per_round)
+            total_produced += num_msg_per_round
+
+    @cluster(num_nodes=4)
+    def test_crash_all(self):
+        """
+        Crash all nodes and restart them. This test should observe data loss
+        and validate the liveness of the system.
+        """
+
+        num_restart_rounds = 2
+        num_msg_per_round = 5000
+
+        total_produced = 0
+        total_lost = 0
+        prev_hwm = 0
+        for round_ix in range(0, num_restart_rounds + 1):
+            self.logger.info(f"Round {round_ix}/{num_restart_rounds}")
+
+            if round_ix > 0:
+                self._crash_and_restart_all()
+                self._wait_for_leader()
+                hwm = next(self.rpk.describe_topic(self.topic)).high_watermark
+                lost = num_msg_per_round - (hwm - prev_hwm)
+                prev_hwm = hwm
+
+                self.logger.info(f"Lost messages: {lost} in round {round_ix}")
+                total_lost += lost
+
+            KgoVerifierProducer.oneshot(self.test_context, self.redpanda,
+                                        self.topic_name, self.MSG_SIZE,
+                                        num_msg_per_round)
+            total_produced += num_msg_per_round
+
+        assert total_lost > 0, "No lost messages observed. The test is not valid."
+
+        # Validate that we can still consume all messages, less lost ones,
+        # and that no funny things happened inside redpanda.
+        final_consumer = KgoVerifierSeqConsumer.oneshot(self.test_context,
+                                                        self.redpanda,
+                                                        self.topic_name,
+                                                        loop=False)
+        assert (total_produced -
+                total_lost == final_consumer.consumer_status.validator.
+                valid_reads), "unexpected number of consumed messages"
+
+    @cluster(num_nodes=4)
+    def test_leader_epoch(self):
```

Comment:
I believe you already added a similar test in unit tests,, wondering if we need this (in the spirit of reducing test runtime).. I don't mind keeping it, just asking.

Reply:
They are similar in the expectations but the behavior is different enough to have a separate test. This one asserts the behavior during data-loss scenario thus exercising more than just kafka layer.